sable_ircd/src/messages/send_history.rs

```diff
@@ -50,13 +50,22 @@ impl SendHistoryItem for update::NewUser {
 }
 
 impl SendHistoryItem for update::UserAwayChange {
-    fn send_to(&self, conn: impl MessageSink, _from_entry: &HistoryLogEntry) -> HandleResult {
+    fn send_to(&self, conn: impl MessageSink, from_entry: &HistoryLogEntry) -> HandleResult {
         if Some(self.user.user.id) == conn.user_id() {
+            // Echo back to the user
             let message = match self.new_reason {
                 None => numeric::Unaway::new(),
                 Some(_) => numeric::NowAway::new(),
             };
             conn.send(message.format_for(&self.user, &self.user));
+        } else {
+            // Tell other users sharing a channel if they enabled away-notify
+            let message = match self.new_reason {
+                None => message::Unaway::new(&self.user),
+                Some(reason) => message::Away::new(&self.user, reason.value()),
+            };
+            let message = message.with_tags_from(from_entry);
+            conn.send(message.with_required_capabilities(ClientCapability::AwayNotify));
         }
```

Comment:
Justification for putting this in an `else` block: https://github.com/ircv3/ircv3-specifications/pull/531